### DISCUSSION TITLE
[frameworks] Flutter Web

### DISCUSSION THREAD

**Discussion by hden** [vercel/vercel#4939] [Ideas] [answered]
## Why?
> Flutter supports the generation of web content rendered using standards-based web technologies: HTML, CSS and JavaScript. -- https://flutter.dev/web

## How?
It's just a static site with unique build commands.

1. Install the flutter SDK
1. Run `flutter build web`
1. Serve the static assets under `/build/web/*`

https://flutter.dev/docs/deployment/web#building-the-app-for-release

**Comment by markozxuu** [comment]
I don't know if this can be used with the builder `@vercel/node `.  This is possible ? cc @TooTallNate @styfle

**Comment by styfle** [comment, marked as answer]
It should be possible already by assigning the Build Command and Output Directory in your Project Settings.

https://vercel.com/blog/advanced-project-settings

**Reply by hden** [reply]
Confirmed that it's working.

The build command can be found here:
https://github.com/flutter/flutter/issues/60853#issuecomment-657795544

**Reply by CodeDoctorDE** [reply]
Is there any plan to add this as a preset to simplify this?

**Reply by connorads** [reply]
Thanks for the link @hden, helped me to get it working.
_I had to adjust it slightly as there is a 256 character limit on the build command since that was posted._

### Build & Development Settings
- Framework Preset: `Other`
- Build Command: `flutter/bin/flutter build web --release`
- Output Directory: `build/web`
- Install Command: `if cd flutter; then git pull && cd .. ; else git clone https://github.com/flutter/flutter.git; fi && ls && flutter/bin/flutter doctor && flutter/bin/flutter clean && flutter/bin/flutter config --enable-web`

![image](https://user-images.githubusercontent.com/10026538/131263924-1d2e06ab-7b0f-4704-9cfa-98cac6f336f5.png)

**Comment by Achudh** [comment]
How do I pass env to flutter? I can do it with the build command but I need to assign different env based on prod, dev, preview @connorads @hden

**Reply by connorads** [reply]
Do the Vercel docs answer your question?
https://vercel.com/docs/concepts/deployments/build-step#environment-variables

**Comment by macwilko** [comment]
If you're having problems building the web app:

One modification to @styfle answer, I recommend, is putting your Flutter version in the git clone:

I am building for Flutter 3.7.12 here, but the git clone will default to the latest stable version (which in my case, was incompatible with some dependencies I'm using).

```bash
if cd flutter; then git pull && cd .. ; else git clone --branch 3.7.12 https://github.com/flutter/flutter.git; fi && ls && flutter/bin/flutter doctor && flutter/bin/flutter clean && flutter/bin/flutter config --enable-web
```

**Reply by anandyadav21219** [reply]
It's helps

**Comment by The-RootCause** [comment]
Any updates concerning Flutter web native support ?
